src/ngx-uploader/classes/ngx-uploader.class.ts

```diff
@@ -26,13 +26,15 @@ export class NgUploaderService {
 
   constructor(concurrency: number = Number.POSITIVE_INFINITY, contentTypes: string[] = ['*']) {
     this.queue = [];
-    this.serviceEvents = new EventEmitter<any>();
+    this.serviceEvents = new EventEmitter<UploadOutput>();
     this.uploadScheduler = new Subject();
     this.subs = [];
     this.contentTypes = contentTypes;
 
     this.uploadScheduler
-      .mergeMap(upload => this.startUpload(upload), concurrency)
+      .pipe(
```

Comment:
hi @DavidRouyer. sorry for super late response. can you explain what is this different from the current implementation? thanks.

Reply:
Check the docs [here](https://github.com/ReactiveX/rxjs/blob/master/doc/pipeable-operators.md). Pipeable operators were introduced in RxJS 5.5 and you use them like this.